src/frontend/src/handler/create_table.rs

```diff
@@ -741,17 +745,150 @@ fn gen_table_plan_inner(
+
+    let pk_column_indices = {
+        let mut id_to_idx = HashMap::new();
+        columns.iter().enumerate().for_each(|(idx, c)| {
+            id_to_idx.insert(c.column_id(), idx);
+        });
+        // pk column id must exist in table columns.
+        pk_column_ids
+            .iter()
+            .map(|c| id_to_idx.get(c).copied().unwrap())
+            .collect_vec()
+    };
+    let table_pk = pk_column_indices
+        .iter()
+        .map(|idx| ColumnOrder::new(*idx, OrderType::ascending()))
+        .collect();
+
+    let connect_properties =
+        derive_connect_properties(source.as_ref(), external_table_name.clone())?;
+
+    let cdc_table_desc = CdcTableDesc {
+        table_id: source.id.into(), // source can be considered as an external table
+        external_table_name: external_table_name.clone(),
+        pk: table_pk,
+        columns: columns.iter().map(|c| c.column_desc.clone()).collect(),
+        stream_key: pk_column_indices,
+        value_indices: (0..columns.len()).collect_vec(),
+        connect_properties,
+    };
+
+    tracing::debug!(?cdc_table_desc, "create cdc table");
+
+    let logical_scan = LogicalScan::create_for_cdc(
+        external_table_name,
+        Rc::new(cdc_table_desc),
+        context.clone(),
+    );
+
+    let scan_node: PlanRef = logical_scan.into();
+    let required_cols = FixedBitSet::with_capacity(columns.len());
+    let mut plan_root = PlanRoot::new(
+        scan_node,
+        RequiredDist::Any,
+        Order::any(),
+        required_cols,
+        vec![],
+    );
+
+    let materialize = plan_root.gen_table_plan(
+        context,
+        name,
+        columns,
+        definition,
+        pk_column_ids,
+        None,
+        append_only,
+        vec![], // no watermarks
+        Some(col_id_gen.into_version()),
+    )?;
+
+    let mut table = materialize.table().to_prost(schema_id, database_id);
+    table.owner = session.user_id();
+    Ok((materialize.into(), None, table))
+}
+
+fn derive_connect_properties(
+    source: &SourceCatalog,
+    external_table_name: String,
+) -> Result<BTreeMap<String, String>> {
+    // we should remove the prefix from `full_table_name`
+    let mut connect_properties = source.properties.clone();
+    if let Some(connector) = source.properties.get(UPSTREAM_SOURCE_KEY) {
+        let table_name = match connector.as_str() {
+            MYSQL_CDC_CONNECTOR => {
+                let db_name = connect_properties.get(DATABASE_NAME_KEY).ok_or_else(|| {
+                    anyhow!("{} not found in source properties", DATABASE_NAME_KEY)
+                })?;
+
+                let prefix = format!("{}.", db_name.as_str());
+                external_table_name
+                    .strip_prefix(prefix.as_str())
+                    .ok_or_else(|| anyhow!("external table name must contain database prefix"))?
+            }
+            #[allow(unused_variables, non_snake_case)]
+            POSTGRES_CDC_CONNECTOR => {
```

Comment:
This is a variable binding instead of a pattern match to the constant. Is that what you expect?